### ISSUE TITLE
Re-export FooMethods and FooCast from DOM module foo

### ISSUE THREAD

**Issue by jdm** [servo/servo#3576]
Instead of making other DOM code import `dom::bindings::codegen::Bindings::FooBinding::FooMethods` in order to call methods on `JSRef<Foo>`, we should just re-export (ie. `pub use`) the FooMethods and FooCast traits for each type in its DOM implementation module. Then users can do `use dom::foo::{Foo, FooMethods, FooCast}` and it will be much more readable.

**Comment by gilles-leblanc** [comment]
@ctlevi Hi, I'm wondering if you're still working on this? Otherwise I would like to take it.

**Comment by jdm** [comment]
@Manishearth actually has a syntax extension that should be able to accomplish this automatically after the next Rust upgrade, so I don't think it's worth investing the time to do this manually any more.

**Comment by Manishearth** [comment]
Yeah, I have one. https://github.com/Manishearth/servo/compare/reexport , but it doesn't work yet because rustup

**Comment by nox** [comment]
Not all interfaces have a method trait.

**Comment by nox** [comment]
We could codegen a module that contains the cast, the constants module and the method trait, and then use * in that from a plugin.

**Comment by jdm** [comment]
We've lived with it this long. I don't think the longer use path is a big deal.
